### PR TITLE
feat(config): add Philio Technology Smart Keypad

### PR DESCRIPTION
**Pull request by MartinHjelmare**
- Add config file for Philio Technology Smart Keypad (psk01).
- Add two templates that could be used frequently for devices from this manufacturer.

### CODE REVIEW

File: packages/config/config/devices/0x013c/psk01.json [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+			"#": "3",
+			"$import": "~/templates/master_template.json#base_enable_disable",
+			"label": "Rand Mode",
+			"description": "When the Rand Mode is enabled, whenever the device wakes up from sleeping, no more than four digits of random keys will light on. Users have to press those random keys one by one, and the device will go to Entry Control mode."
+		},
+		{
+			"#": "4",
+			"$import": "~/templates/master_template.json#base_enable_disable",
+			"label": "Touch Sleep",
+			"description": "If Touch Sleep is enabled the device will go to sleep a short time after the last touch.",
```
Comment (MartinHjelmare) [inline packages/config/config/devices/0x013c/psk01.json]: The device doesn't actually go to sleep it seems. It's just the panel lights that turn off until the next time the panel is touched. The manual doesn't explain the details.
```suggestion
			"description": "If Touch Sleep is enabled the touch panel lights will turn off a short time after the last touch.",
```

File: packages/config/config/devices/0x013c/psk01.json [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+	"firmwareVersion": {
+		"min": "0.0",
+		"max": "255.255"
+	},
+	"paramInformation": [
+		{
+			"#": "1",
+			"$import": "~/templates/master_template.json#base_options_nounit",
+			"label": "Set Keypad Mode",
+			"description": "Change between Entry control mode and Central Scene mode.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x013c/psk01.json]:
```suggestion
```
The options already imply this.

File: packages/config/config/devices/0x013c/psk01.json [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+					"label": "Central Scene mode",
+					"value": 2
+				}
+			]
+		},
+		{
+			"#": "2",
+			"$import": "~/templates/master_template.json#base_enable_disable",
+			"label": "Buzzer",
+			"description": "Enable or disable the buzzer.",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x013c/psk01.json]:
```suggestion
```
Dito

File: packages/config/config/devices/0x013c/psk01.json [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+		{
+			"#": "5",
+			"$import": "templates/philio_template.json#auto_report_tick",
+			"label": "Auto Report Time Unit",
+			"description": "Define the time scale for automatic report. This setting will affect configuration no. 6."
+		},
+		{
+			"#": "6",
+			"$import": "templates/philio_template.json#auto_report_interval",
+			"label": "Auto Report Battery Time",
```
Comment (AlCalzone) [inline packages/config/config/devices/0x013c/psk01.json]: We typically call this

```suggestion
			"label": "Battery Report Interval",
```

File: packages/config/config/devices/0x013c/psk01.json [hunk 5]
```diff
@@ -0,0 +1,73 @@
...
+			"description": "The time interval upon which to report the battery level."
```
Comment (AlCalzone) [inline packages/config/config/devices/0x013c/psk01.json]:
```suggestion
```
With a good label, the description is unnecessary.

File: packages/config/config/devices/0x013c/psk01.json [hunk 6]
```diff
@@ -0,0 +1,73 @@
...
+			"$import": "~/templates/master_template.json#base_enable_disable",
+			"label": "Touch Sleep",
+			"description": "If Touch Sleep is enabled the touch panel lights will turn off a short time after the last touch.",
+			"defaultValue": 1
+		},
+		{
+			"#": "5",
+			"$import": "templates/philio_template.json#auto_report_tick",
+			"label": "Auto Report Time Unit",
+			"description": "Define the time scale for automatic report. This setting will affect configuration no. 6."
```
Comment (AlCalzone) [inline packages/config/config/devices/0x013c/psk01.json]: I don't understand this parameter. Is it a multiplier for the interval set in parameter 6?

**Review by AlCalzone — CHANGES_REQUESTED** [review]

File: packages/config/config/devices/0x013c/psk01.json (hunk 6 above)
Comment (MartinHjelmare) [reply packages/config/config/devices/0x013c/psk01.json]: Yes, exactly.

Comment (MartinHjelmare) [reply packages/config/config/devices/0x013c/psk01.json]:
```suggestion
			"$import": "templates/philio_template.json#auto_report_tick",
			"label": "Auto Report Time Unit",
			"description": "Define the time scale for automatic reports. This value will be multiplied with configuration no. 6. to set the time interval between reports."
```

**Review by AlCalzone — APPROVED** [review]
This is such an odd parameter design...

File: packages/config/config/devices/0x013c/templates/philio_template.json [hunk 7]
```diff
@@ -36,10 +36,12 @@
 			}
 		]
 	},
-	"auto_report_interval": {
+	"battery_report_interval_30m": {
```
Comment (MartinHjelmare) [inline packages/config/config/devices/0x013c/templates/philio_template.json:39]: The point with the new templates was that it's not only battery that is reported automatically with this pattern. I was planning to use this for all auto reports. I don't think this change was good.

Comment (AlCalzone) [reply packages/config/config/devices/0x013c/templates/philio_template.json:39]: I see. Will rename it then.

**Pull request merged**